x-pack/plugins/licensing/server/licensing.test.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { take, skip } from 'rxjs/operators';
+import { merge } from 'lodash';
+import { ClusterClient } from 'src/core/server';
+import { coreMock } from '../../../../src/core/server/mocks';
+import { ILicensingPluginSetup } from './types';
+import { LicensingPlugin } from './licensing_plugin';
+import { LicensingPluginSetup } from './licensing_plugin_setup';
+import { LICENSE_STATUS } from './constants';
+
+interface TestServices {
+  plugin: LicensingPlugin;
+  service$: BehaviorSubject<ILicensingPluginSetup>;
+  licensing: ILicensingPluginSetup;
+  clusterClient: jest.Mocked<PublicMethodsOf<ClusterClient>>;
+}
+
+describe('licensing', () => {
```

Comment:
Can we split up these tests? I think each file should have it's own file of unit tests that are pretty isolated.

Reply:
These tests are all for the setup license interface, all a single file.